.github/workflows/release.yaml

```diff
@@ -1,27 +1,98 @@
-name: Release Charts
+name: Release Docker
 
 on:
   push:
+    tags: 
+      - 'v*.*.*'
     branches:
-      - master
+      - 'master'
 
+env:
+  REGISTRY: quay.io
+  IMAGE_NAME: prometheusmsteams/prometheus-msteams
+  
 jobs:
-  release:
+  build:
+
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@79abd3f86f79a9d68a23c75a09a9a85889262adf
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Log into registry ${{ env.REGISTRY }}
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.DOCKER_QUAY_USER }}
+          password: ${{ secrets.DOCKER_QUAY_TOKEN }}
+
+# build
+
+      - name: Build binaries
+        id: build-binary
+        run: |
+          make all
+
+      - name: Build Docker image
+        id: build-docker
+        uses: docker/build-push-action@ac9327eae2b366085ac7f6a2d02df8aa8ead720a
         with:
-          fetch-depth: 0
+          context: .
+          load: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+# tests
 
-      - name: Configure Git
+      - name: Run tests
+        id: test-binary
         run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          make test
+
+      - name: Test docker image
+        id: test-docker
+        # kudos: https://stackoverflow.com/questions/63641822/run-command-with-timeout-in-github-workflow/63643845?noredirect=1#comment112574154_63643845
+        # keep docker image running for 2 minutes. It should timeout (error code 124).
+        run: | 
+          timeout 120 docker run --rm $(echo ${{ steps.meta.outputs.tags }}|head -n1) || code=$?
+          if [[ $code -ne 124 ]]; then
+            echo "Docker image stopped with ${code}. It was expected to run forever." 
+            exit 1;
+          fi
+
+# deploy releases
+
+      - name: Create github release
+        id: github-release
+        uses: softprops/action-gh-release@v1
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          fail_on_unmatched_files: true
+          generate_release_notes: true
+          files: |
```

Comment:
Do you think it's possible to have the corresponding Helm Chart here as well? I currently find it very confusing looking at the release page that we have releases for the app itself and for the Helm chart separately.

Reply:
@Knappek it's an issue with the current workflow.

The Pullrequest which bumps the helm chart version + docker image tag is failing during the "install test" step.
The reason is simple: the docker image does not exist at this time.

The current workflow therefore enforces to create the docker image first by tagging master and afterwards the version can be bumped in the helm-chart and docs can be updated through another PR. This creates the helm release.
The helm chart version is not available at the time of building the docker image.